src/opnmidi_midiplay.cpp

```diff
@@ -261,7 +298,7 @@ bool OPNMIDIplay::realTime_NoteOn(uint8_t channel, uint8_t note, uint8_t velocit
 
     if(static_cast<size_t>(channel) > m_midiChannels.size())
         channel = channel % 16;
-    noteOff(channel, note);
+    noteOff(channel, note, true);
```

Comment:
```cpp
noteOff(channel, note, (velocity != 0));
```
Otherwise, the `War3#12.mid` will be broken. And any other MIDI files are having "NoteOn with zero velocity" as alternative form of "NoteOff" event.

Reply:
Good catch, yes; fixed it here and also in libADLMIDI.